OrbitCore/LinuxEventTracer.cpp

```diff
@@ -172,13 +193,89 @@ void LinuxEventTracer::Run(bool* a_ExitRequested)
+                            fds.push_back(fd);
+                            ring_buffers.emplace_back(fd);
+                            LinuxPerfUtils::start_capturing(fd);
+                        }
+                    }
+                    break;
+
+                case PERF_RECORD_EXIT:
+                    {
+                        auto exit = ring_buffer.ConsumeRecord<LinuxForkEvent>(header);
+                        if (exit.PID() == a_Pid) {
+                            // TODO: stop event recording, close ring buffer, close file descriptor.
+                        }
+                    }
+                    break;
+
+                case PERF_RECORD_MMAP:
+                    {
+                        // There was a call to mmap with PROT_EXEC, hence
+                        // refresh the maps. This should happen rarely.
+                        ring_buffer.SkipRecord(header);
+                        unwinder.SetMaps(LinuxUtils::ReadMaps(a_Pid));
+                    }
+                    break;
+
+                case PERF_RECORD_SAMPLE:
+                    {
+                        auto sample = ring_buffer.ConsumeRecord<LinuxStackSampleEvent>(header);
+
+                        std::vector<unwindstack::FrameData> frames =
+                            unwinder.Unwind(sample.Registers(), sample.StackDump(), sample.StackSize());
+                        if (frames.empty()) {
+                            continue;
+                        }
+
+                        CallStack cs;
+                        cs.m_ThreadId = sample.TID();
+                        for (const auto& frame : frames) {
+                            std::wstring moduleName = ToLower(Path::GetFileName(s2ws(frame.map_name)));
+                            std::shared_ptr<Module> moduleFromName = Capture::GTargetProcess->GetModuleFromName( ws2s(moduleName) );
+
+                            uint64_t address = frame.pc;
+                            if (moduleFromName) {
+                                uint64_t new_address = moduleFromName->ValidateAddress(address);
+                                address = new_address;
+                            }
+                            cs.m_Data.push_back(address);
+
+                            if( Capture::GTargetProcess &&
+                                    !Capture::GTargetProcess->HasSymbol(address) &&
+                                    !frame.function_name.empty())
+                            {
+                                std::stringstream ss;
+                                ss << LinuxUtils::Demangle(frame.function_name.c_str()) << "+0x" << std::hex << frame.function_offset;
+                                GCoreApp->AddSymbol(address, frame.map_name, ss.str());
+                            }
+                        }
+                        cs.m_Depth = cs.m_Data.size();
+
+                        // This is what LinuxPerf::HandleLine is doing,
```

Comment:
I don't understand this comment. Also, I expect LinuxPerf will be removed soon-ish and then this comment will be stale. Rather comment why this is the right thing to do.